dynamodb_feature_store.js

```diff
@@ -59,7 +59,7 @@ function dynamoDBFeatureStoreInternal(tableName, options) {
         }
       }
       cb(results);
-    }, function (err) {
+    }).catch(function (err) {
```

Comment:
The issue here is that if you do `somePromise.then(successHandler, errorHandler)`, then `errorHandler` will be called if `somePromise` was rejected— but _not_ if `successHandler` throws an exception or returns a rejected promise. In order for the latter to work, you have to do `somePromise.then(successHandler).catch(errorHandler)` instead.